src/Tasks/BootstrapperUtil/BootstrapperBuilder.cs

```diff
@@ -444,13 +444,20 @@ public string[] GetOutputFolders(string[] productCodes, string culture, string f
 
             foreach (string file in files)
             {
-                string folder = System.IO.Path.GetDirectoryName(file);
-                if (folder.Substring(0, invariantPath.Length).ToLowerInvariant().CompareTo(invariantPath) == 0)
+                List<string> packagePaths = new List<string>() { invariantPath };
+                packagePaths.AddRange(Util.AdditionalPackagePaths.Select(p => Util.AddTrailingChar(p.ToLowerInvariant(), System.IO.Path.DirectorySeparatorChar)));
+                foreach (string packagePath in packagePaths)
                 {
-                    string relPath = folder.Substring(invariantPath.Length);
-                    if (!folders.Contains(relPath))
+                    string folder = System.IO.Path.GetDirectoryName(file);
+                    if (folder.Length >= packagePath.Length && folder.Substring(0, packagePath.Length).ToLowerInvariant().CompareTo(packagePath) == 0)
                     {
-                        folders.Add(relPath);
+                        string relPath = folder.Substring(packagePath.Length);
+                        if (!folders.Contains(relPath))
```

Comment:
Is the logic still correct? We might get the same relative path from multiple packages and I am not sure the results will be used correctly.

Reply:
Yes, this is still correct.  This method is used to determine the additional subfolders needed in the click once publish directory when you have the "Download prerequisites from the same location as my application" option selected.  The relative path comes from the relative path of the package under the packages directory (Example: "C:\Program Files (x86)\Microsoft SDKs\ClickOnce Bootstrapper\Packages\DotNetFX472" => "DotNetFX472").  In the unlikely case where an identical sub folder name is used between two packages that come from different package folders the package files from both would just get copied into this same subfolder.

Reply:
Sounds good.